docker/transport/sshconn.py

```diff
@@ -72,13 +74,25 @@ def _get_conn(self, timeout):
 class SSHHTTPAdapter(BaseHTTPAdapter):
 
     __attrs__ = requests.adapters.HTTPAdapter.__attrs__ + [
-        'pools', 'timeout', 'ssh_client',
+        'pools', 'timeout', 'ssh_client', 'ssh_conf'
     ]
 
     def __init__(self, base_url, timeout=60,
                  pool_connections=constants.DEFAULT_NUM_POOLS):
+        logging.getLogger("paramiko").setLevel(logging.WARNING)
         self.ssh_client = paramiko.SSHClient()
+        self.ssh_conf = {}
+        ssh_config_file = os.path.expanduser("~/.ssh/config")
+        if os.path.exists(ssh_config_file):
+            conf = paramiko.SSHConfig()
+            with open(ssh_config_file) as f:
+                conf.parse(f)
+            parsed = six.moves.urllib_parse.urlparse(base_url)
+            host_config = conf.lookup(parsed.hostname)
+            self.ssh_conf = host_config
+
         self.ssh_client.load_system_host_keys()
+        self.ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
```

Comment:
If we aren't going to obey the various config options which effect this (`StrictHostKeyChecking` et al) then I think `WarningPolicy()` would be a better one to hard code.

Reply:
I changed this in 0e2fe7f